internal/api/courses/stats/query_test.go

```diff
@@ -10,34 +10,113 @@ import (
+		email           string
+		expectedLocator string
+		query           *stats.Query
+		expectedValues  []int
 	}{
-		// Base
-		{"server-admin", "", stats.CourseMetricQuery{}, []int{100, 200, 300}},
-		{"server-admin", "", stats.CourseMetricQuery{BaseQuery: stats.BaseQuery{Sort: 1}}, []int{300, 200, 100}},
-		{"server-admin", "", stats.CourseMetricQuery{BaseQuery: stats.BaseQuery{After: timestamp.FromMSecs(150)}}, []int{200, 300}},
-
-		// Course Specific
-		{"server-admin", "", stats.CourseMetricQuery{AssignmentID: "A2"}, []int{200}},
-		{"server-admin", "", stats.CourseMetricQuery{AssignmentID: "ZZZ"}, nil},
-		{"server-admin", "", stats.CourseMetricQuery{UserEmail: "U1"}, []int{100, 200}},
-		{"server-admin", "", stats.CourseMetricQuery{Type: stats.CourseMetricTypeGradingTime}, []int{100, 300}},
-
-		// Error
-		{"server-user", "-040", stats.CourseMetricQuery{}, nil},
-		{"course-student", "-020", stats.CourseMetricQuery{}, nil},
+		// Grading Time Stats.
+		{
+			email: "course-admin",
+			query: &stats.Query{
+				Type: stats.Grading_Time_Stats_Type,
+				Where: map[stats.MetricAttribute]any{
+					stats.Course_ID_Key: db.TEST_COURSE_ID,
+				},
+			},
+			expectedValues: []int{100},
+		},
+		{
+			email: "course-admin",
+			query: &stats.Query{
+				Type: stats.Grading_Time_Stats_Type,
+				Where: map[stats.MetricAttribute]any{
+					stats.Course_ID_Key: "zzz",
+				},
+			},
+			expectedValues: []int{100},
+		},
+
+		// Task Time Stats.
+		{
+			email: "course-admin",
+			query: &stats.Query{
+				Type: stats.Task_Time_Stats_Type,
+				Where: map[stats.MetricAttribute]any{
+					stats.Course_ID_Key: db.TEST_COURSE_ID,
+				},
+			},
+			expectedValues: []int{100},
+		},
+		{
+			email: "course-admin",
+			query: &stats.Query{
+				Type: stats.Task_Time_Stats_Type,
+				Where: map[stats.MetricAttribute]any{
+					stats.Course_ID_Key: "zzz",
+				},
+			},
+			expectedValues: []int{100},
+		},
+
+		// Code Analysis Time Stats
+		{
+			email: "course-admin",
+			query: &stats.Query{
+				Type: stats.Code_Analysis_Time_Stats_Type,
+				Where: map[stats.MetricAttribute]any{
+					stats.Course_ID_Key: db.TEST_COURSE_ID,
+				},
+			},
+			expectedValues: []int{100},
+		},
+		{
+			email: "course-admin",
+			query: &stats.Query{
+				Type: stats.Code_Analysis_Time_Stats_Type,
+				Where: map[stats.MetricAttribute]any{
+					stats.Course_ID_Key: "zzz",
+				},
+			},
+			expectedValues: []int{100},
+		},
+
+		// Grading Time Stats, Error
+		{
+			email:           "course-user",
+			expectedLocator: "-017",
+			query: &stats.Query{
+				Type: stats.Grading_Time_Stats_Type,
+			},
+		},
+
+		// Task Time Stats, Error
+		{
+			email:           "course-user",
+			expectedLocator: "-017",
+			query: &stats.Query{
+				Type: stats.Task_Time_Stats_Type,
+			},
+		},
+
+		// Code Analysis Time Stats, Error
+		{
+			email:           "course-user",
+			expectedLocator: "-017",
+			query: &stats.Query{
+				Type: stats.Task_Time_Stats_Type,
+			},
```

Comment:
These cases look strange.
Why -017 (no password)?
Are you trying for a perm error?
If so, why bother with the different query types?

The "Error" in the comment is not useful.
What type of error?